components/collapsible-panel/collapsible-panel-group.js

```diff
@@ -32,39 +27,32 @@ class CollapsiblePanelGroup extends SkeletonGroupMixin(LitElement) {
 
 	constructor() {
 		super();
-		this._panels = [];
+		this._panels = new SubscriberRegistryController(this, 'collapsible-panel-group');
 	}
 
 	render() {
 		const classes = {
 			spaced: this._panels?.[0]?.type !== 'inline',
 		};
 
-		return html`<slot class="${classMap(classes)}" @slotchange="${this._handleSlotChange}"></slot>`;
-	}
-
-	async _getPanels() {
-		const slot = this.shadowRoot?.querySelector('slot');
-		if (!slot) return;
-
-		return slot.assignedNodes({ flatten: true }).filter((node) => node.nodeType === Node.ELEMENT_NODE);
-	}
-
-	async _handleSlotChange() {
-		this._panels = await this._getPanels();
-		this._updatePanelAttributes();
+		return html`<slot class="${classMap(classes)}"></slot>`;
 	}
 
 	_updatePanelAttributes() {
-		if (!this._panels || this._panels.length === 0) return;
-
-		if (this._panels[0]?.type !== 'inline') return;
-
-		for (const panel of this._panels) {
-			panel._noBottomBorder = true;
+		const panels = this.shadowRoot
+			?.querySelector('slot')
+			?.assignedNodes({ flatten: true })
+			.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName === 'D2L-COLLAPSIBLE-PANEL');
+		if (!panels || panels.length === 0) return;
```

Comment:
I see a lot of `x.length === 0` checks around. Is there a reason for this over `if (x.length)`?

```suggestion
		if (panels?.length) return;
```